src/plugins/contextMenu/contextMenu.js

```diff
@@ -236,6 +236,45 @@ class ContextMenu extends BasePlugin {
     this.commandExecutor.execute.apply(this.commandExecutor, params);
   }
 
+  changeItems(removeItemFunction, newItems) {
+
+    if (typeof removeItemFunction === 'function') {
+      arrayEach(Object.entries(this.itemsFactory.predefinedItems), (entry) => {
+        let key = entry[0];
+        let item = entry[1];
+        if (removeItemFunction(key, item)) {
+          delete this.itemsFactory.predefinedItems[key];
+        }
+      });
+    }
+
+    arrayEach(Object.entries(newItems || {}), (entry) => {
```

Comment:
the second param is an object containing all of the new items that we wish to add to the context menu. Note that the remove function is called first, in case any new items added here happen to match the remove function filter.